sympy/solvers/tests/test_solveset.py

```diff
@@ -683,12 +683,15 @@ def test_solve_quintics():
 def test_solveset_complex_exp():
     from sympy.abc import x, n
     assert solveset_complex(exp(x) - 1, x) == \
-        imageset(Lambda(n, I*2*n*pi), S.Integers)
```

Comment:
How does intersecting with `S.Complexes` help, `I*2*n*pi` is already a complex number.